porcupine/plugins/comment_selected_lines.py

```diff
@@ -22,8 +31,17 @@ def comment_or_uncomment(tab: tabs.FileTab, pressed_key: str | None = None) -> s
     try:
         start_index, end_index = map(str, tab.textwidget.tag_ranges("sel"))
     except ValueError:
-        # nothing selected, add '#' normally
-        return None
+        # No text selected
+        lineno = tab.textwidget.index("insert").split(".")[0]
+        line_start = lineno + ".0"
+        line_text = tab.textwidget.get(line_start, f"{line_start} lineend")
+
+        if already_commented(line_text, comment_prefix):
+            tab.textwidget.delete(line_start, f"{lineno}.{len(comment_prefix)}")
+        else:
+            tab.textwidget.insert(f"{lineno}.0", comment_prefix)
```

Comment:
It may be a big ask but what if we made all the comments look like `# blah` by finding the indents?

Reply:
My first version of this code did look for the indentation and put the comment syntax one char before first indentation level. Similar to how it works in VS Code. For example

```
def foo():
    lol = 1
    if lol == 1:
        print("bar")
        print("baz")
```

If I commented out line 3 & 4 of this block it would be:

```
def foo():
    lol = 1
    # if lol == 1:
    #    print("bar")
    #    print("baz")
```

This is because the following would look really messy:

```
def foo():
    lol = 1
    # if lol == 1:
       # print("bar")
       # print("baz")
```

However, Akuli requested that I keep all comment syntax pushed to the left.

Reply:
@Akuli is right :)

Reply:
@rdbende 

With the current state of this PR, the comments look like

```
def foo():
    lol = 1
    if lol == 1:
        if True:
            if True:
#                print("bar")
#                print("baz")
```

The same comment in for example VS Code would look like

```
def foo():
    lol = 1
    if lol == 1:
        if True:
            if True:
            #        print("bar")
            #        print("baz")
```

To clarify, do you think column 0 is better than the indentation level of what is commented?

Reply:
No, I think that comments should be at the indentation level of the outermost block

So this:
```py
def foo():
    lol = 1
    if lol == 1:
        # if True:
        #     if True:
        #         print("bar")
        #         print("baz")
```

instead of this:

```py
def foo():
    lol = 1
    if lol == 1:
        # if True:
            # if True:
                # print("bar")
                # print("baz")
```

Reply:
Black will change it to this anyway :)

```py
def foo():
    lol = 1
    if lol == 1:
        # if True:
        # if True:
        # print("bar")
        # print("baz")
        ...
```

Reply:
Ah, right. So to clarify

My first version of the code added comment syntax at the indentation level of the outermost block (never published PR). However, Akuli wanted it like follows:

```
def foo():
    lol = 1
    if lol == 1:
#         if True:
#             if True:
#                 print("bar")
#                 print("baz")
```

Reply:
Ah, I see. Then I don't agree woth Akuli. Black will indent that anyway, so it doesn't make sense to me.